app/app.ts

```diff
@@ -150,11 +150,11 @@ function setupIpc(localSettings: LocalSettings, scrSettings: ScrSettings) {
     })
 
   localSettings.on('change', settings => {
-    if (settings.runAppAtSystemStart !== app.getLoginItemSettings().openAtLogin) {
-      app.setLoginItemSettings({
-        openAtLogin: settings.runAppAtSystemStart,
-      })
-    }
+    app.setLoginItemSettings({
+      openAtLogin: settings.runAppAtSystemStart,
+      args: [settings.runAppAtSystemStartMinimized ? '--hidden' : ''],
+    })
+
```

Comment:
@2Pacalypse- I removed the check to see if runAppAtSystemStart changed since I didn't see a simple way to also check if --hidden was already in the args. It seems to be in getLoginSettings().launchItems[].args. I'm not sure how I would determine which of the launchItems[] to check args for. I can look into it more if you'd like the checks back.

Reply:
Yeah I would still like to have the check on, but maybe a better way to write it would be to save the last value of these two values outside this handler, and then check against them. Something like:
```ts
let lastRunAppAtSystemStart: boolean | undefined
let lastRunAppAtSystemStartMinimized: boolean | undefined

localSettings.on('change', settings => {
  if (
    lastRunAppAtSystemStart !== settings.runAppAtSystemStart ||
    lastRunAppAtSystemStartMinimized !== settings.runAppAtSystemStartMinimized
  ) {
    app.setLoginItemSettings({
      openAtLogin: settings.runAppAtSystemStart,
      args: [settings.runAppAtSystemStartMinimized ? '--hidden' : ''],
    })
    lastRunAppAtSystemStart = settings.runAppAtSystemStart
    lastRunAppAtSystemStartMinimized = settings.runAppAtSystemStartMinimized
  }

  //
```

Reply:
I made this change, pretty much verbatim to what you have here. It works as expected and it seems the best way to do it.